src/openforms/utils/json_logic.py

```diff
@@ -49,3 +52,174 @@ def normalize_value(value: Union[dict, list, tuple, int, str]) -> JSONLogicValue
+    :return: rule with additional informations related to the involved component
+
+
+    """
+
+    if isinstance(logics, JsonLogicTest):
+        logic_dict = {
+            "nodeType": "operation",
+            "operator": logics.operator,
+            "children": [],
+        }
+        for index, value in enumerate(logics.values):
+
+            # There is an operation
+            if isinstance(value, JsonLogicTest):
+                if value.operator in ["var", "date"]:
+                    key = find_json_logic_test_value(logics.values[index])
+                    # leaf not related to component --> just a value compared
+                    if not component_map.get(key):
+                        logic_dict["value"] = key
+
+                    # Leaf is a field
+                    else:
+                        leaf_node = {
+                            "nodeType": "leaf",
+                            "key": key,
+                            "actualValue": resulting_data.get(key, ""),
+                            "stepName": component_map[
+                                key
+                            ].form_step.form_definition.name,
+                            "label": component_map[key].component.get("label", ""),
+                        }
+                        logic_dict["children"].append(leaf_node)
+                elif value.operator == "rdelta":
+                    logic_dict["value"] = value.values
+
+                # No relevant information with this operator
+                elif value.operator == "today":
+                    pass
+                else:
+                    logic_dict["children"].append(
+                        convert_json_logic_in_json_tree(
+                            value,
+                            component_map,
+                            resulting_data,
+                        )
+                    )
+
+            # No operation, it's a basic value
+            else:
+                logic_dict["value"] = value
+
+        return logic_dict
+
+    # array of rules
+    else:
+        return [
+            convert_json_logic_in_json_tree(logic, component_map, resulting_data)
+            for logic in logics
+        ]
+
+
+def stringify_json_tree(node: Dict[Any, str]):
+    """transform the json_tree in a string of rules"""
+
+    # Unary operation
+    if not node["children"]:
+        return f"{node['operator']} {node['value']}"
+
+    tmp_string = ""
+    children_number = len(node["children"]) - 1
+    for index, child in enumerate(node["children"]):
+
+        if child["nodeType"] == "leaf":
+            tmp_string += child["key"]
+        else:
+            tmp_string += f"({stringify_json_tree(child)})"
+
+        if node.get("value"):
+            tmp_string += f" {node['operator']} {node['value']}"
+        else:
+            print()
+            tmp_string += (
+                f"""{f" {node['operator']} " if index != children_number else ''}"""
+            )
```

Comment:
```suggestion
            output += f" {node['operator']} " if not last else ""
```
I think? I don't see the point of the outer f-string (triple quoted)